test/login_test.exs

```diff
@@ -97,6 +97,42 @@ defmodule LoginTest do
     assert {:ok, %Postgrex.Result{}} = P.query(pid, "SELECT 123", [])
   end
 
+  test "env var default db name" do
+    previous_db_name = System.get_env("PGDATABASE")
+    try do
+      set_db_name("postgrex_test")
+      opts = []
+      assert {:ok, pid} = P.start_link(opts)
+      assert {:ok, %Postgrex.Result{}} = P.query(pid, "SELECT 123", [])
+    after
+      set_db_name(previous_db_name)
+    end
+  end
+
+  test "env var default db name (no such database)" do
+    previous_db_name = System.get_env("PGDATABASE")
+    try do
+      set_db_name("doesntexist")
+      Process.flag(:trap_exit, true)
+
+      capture_log fn ->
+        opts = [ sync_connect: true, backoff_type: :stop ]
+        assert {:error, {%Postgrex.Error{postgres: %{code: :invalid_catalog_name}}, [_|_]}} =
+               P.start_link(opts)
+
+        assert_receive {:EXIT, _, {%Postgrex.Error{postgres: %{code: :invalid_catalog_name}}, [_|_]}}
+      end
+
+      capture_log fn ->
+        opts = [ backoff_type: :stop ]
+        {:ok, pid} = P.start_link(opts)
+        assert_receive {:EXIT, ^pid, {%Postgrex.Error{postgres: %{code: :invalid_catalog_name}}, [_|_]}}
+      end
+    after
```

Comment:
We only need to test this once, either with or without `:sync_connect`. I think the `:sync_connect example is clearer as we would only needs to assert on the `start_link` return value. No need to check for the `:EXIT`.